nixos/modules/installer/sd-card/sd-image.nix

```diff
@@ -201,23 +209,44 @@ in
         imageSize=$((rootSizeBlocks * 512 + firmwareSizeBlocks * 512 + gap * 1024 * 1024))
         truncate -s $imageSize $img
 
+        visionFive=${lib.boolToString config.sdImage.visionFive}
```

Comment:
I'm a little bit wary of adding this firmware-specific hack to mainline nixpkgs, since I expect this to be fixed in the future releases of the vendor U-Boot. At the very least some explanation should be added (https://github.com/starfive-tech/u-boot/issues/30).

Reply:
That's one of the reasons I made this change it's own commit as well, so we could easily revert it.
It is needed, but I'll add some explanation like you ask.